invenio_rdm_records/records/providers/base.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2021 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""PID Base Provider."""
+
+from invenio_pidstore.models import PIDStatus
+
+PID_PROVIDER_EXTERNAL = "external"
+
+
+class Client:
+    """PID Client base class."""
+
+    name = ""
+
+
+class Provider:
+    """Base Provider class."""
+
+    def __init__(self, url, name, client=None, **kwargs):
+        """Constructor."""
+        self.url = url
+        self.name = name
+        self.client = client
+
+    def create(self, status=PIDStatus.RESERVED, **kwargs):
```

Comment:
I am not 100% sure about the `create` and `reserve` signatures.
Is the `status` needed for `create`?

Reply:
Will take it into account in a future refactor, try to use the existing provider API fo pidstore.